js/piwik.js

```diff
@@ -4023,9 +4023,13 @@ if (typeof window.Piwik !== 'object') {
                     if (isFunction(windowAlias.GearsFactory)) {
                         browserFeatures.gears = '1';
                     }
-
-                    // other browser features
-                    browserFeatures.cookie = hasCookies();
+                    
+                    if (!isDefined(windowAlias.showModalDialog) && isDefined(navigatorAlias.cookieEnabled)) {
+                        browserFeatures.cookie = navigatorAlias.cookieEnabled ? '1' : '0';
+                    } else {
+                        // Eg IE11 ... prevent error when cookieEnabled is requested within modal dialog. see #11507
+                        browserFeatures.cookie = hasCookies();
+                    }
```

Comment:
@sgiehl yes it is the same code and kind of copied from there... we can't reuse the same method there though as it's different behaviour. 

Basically here we want to detect if cookies are generally enabled unless it is IE11 or older. for IE 11 and older we need to execute different logic. It would end up in complicated code to put it in the method itself therefore rather copying code so it's in the end easier... so for IE 11 we might report cookies are disabled when cookies are disabled in Matomo and that's fine. But for all other browsers when cookies are disabled we still report the correct value.

Reply:
User tested and confirms it works: https://github.com/matomo-org/matomo/issues/13056#issuecomment-648805386

Except for IE 11 where we will follow up in https://github.com/matomo-org/matomo/pull/16113

Reply:
Can't see any complexity here 🤷 The only difference is about to ignore the check for `configCookiesDisabled`. So it would be basically the same to change the hasCookies method like that:

```php
 function hasCookies(ignoreConfig) { 
     if (ignoreConfig !== true && configCookiesDisabled) { 
         return '0'; 
     } 
  
     if(!isDefined(windowAlias.showModalDialog) && isDefined(navigatorAlias.cookieEnabled)) { 
         return navigatorAlias.cookieEnabled ? '1' : '0'; 
     } 
```

and then call 
```
browserFeatures.cookie = hasCookies(true);
```

Reply:
Config should not be ignored for IE eg when `windowAlias.showModalDialog` is defined.

Reply:
Ok. Guess simply switching the code blocks in `hasCookies` would cause an issue if hasCookies is called eg here: 
https://github.com/matomo-org/matomo/blob/de5ae85e7a11f7bbd962256c890a32188c40d000/js/piwik.js#L3296
Right? If that's the case, should be good to merge...